test/components/attribute-picker/attribute-picker-item.test.js

```diff
@@ -0,0 +1,19 @@
+import '../../../src/components/attribute-picker/attribute-picker-item.js';
+import { fixture, html, oneEvent, runConstructor } from '@brightspace-ui/testing';
+
+describe('attribute-picker-item', () => {
+	describe('constructor', () => {
+		it('should construct', () => {
+			runConstructor('d2l-labs-attribute-picker-item');
+		});
+	});
+
+	describe('eventing', () => {
+		it('should fire d2l-labs-attribute-picker-item-delete when delete button is clicked', async() => {
+			const component = await fixture(html`<d2l-labs-attribute-picker-item text="attribute" deletable></d2l-labs-attribute-picker-item>`);
+			const listener = oneEvent(component, 'd2l-labs-attribute-picker-item-deleted');
+			component.shadowRoot.querySelector('d2l-icon').click();
+			await listener;
```

Comment:
Not sure how much you want to fix vs. just migrate over as-is, but the proper way to do this would be:

```suggestion
			clickElem(component.shadowRoot.querySelector('d2l-icon'));
			await oneEvent(component, 'd2l-labs-attribute-picker-item-deleted');
```

Reply:
I don't mind making the switch over to all of those!

Reply:
Okay, so I updated the tests to use the helpers where possible, with the exception of [here](https://github.com/BrightspaceUI/labs/blob/clabatt/ARK-639-Migrate-Attribute-Picker/test/components/attribute-picker/attribute-picker.vdiff.js#L121) where `clickElem` doesn't seem to get registered (regardless of if it's awaited or not). I also realized that there's no helper for `blur`, but it's not the end of the world.

Reply:
Hmm, I don't love that `<d2l-icon role="button">` business since it's not going to be keyboard accessible. Would converting it to `<d2l-button-icon>` be terrible?

Yeah for blur typically I'll just do `clickAt(0, 0)`.

Reply:
So, we had to do some magic a while ago, but it's actually keyboard accessible, just not in the traditional way, partially because of how things with the `application` role function. Hmm, I can look into using the button instead, I think we had issues with it initially because of it's sizing spacing compared to item. I can revisit that on Monday and talk with my team about making that change

Reply:
So after messing around with the button, there were quire a few complications with using the button icon instead. The main ones where the size of the icon, which didn't seem like we could adjust in the way we wanted, it broke the navigation between the items, and also we couldn't show the hover state when the item wasn't selected, meaning they wouldn't know they could delete the item.

I do think the current solution is essentially fully accessible with the main potential exceptions being if the hover state of the icons is different enough. Would you be okay with things staying as they are currently since it's a labs component? And then if it does look like it'll be promoted to something bigger, than we make those improvements then?

Reply:
Thanks for looking into it. Yeah that's totally fine, although possibly a good middle ground solution would just be to use `<button aria-label="whatever"><d2l-icon ...>` and then style the button & icon however you want. Then you'll get the right keyboard/focus accessibility for free.

Reply:
While working on that I made a realization that I was being silly with the `d2l-button-icon`, I'd used the same icon (`close-large-thick`) instead of switching to `close-small`, which fits a lot more appropriately. So I'm back to trying to get that to work!

Reply:
Had to make a few modifications, but was able to switch to using `d2l-button-icon` along with make a few other improvements, so it should be good to go now 😄